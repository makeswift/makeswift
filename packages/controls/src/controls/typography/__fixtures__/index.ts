export * from './typography-without-id'
