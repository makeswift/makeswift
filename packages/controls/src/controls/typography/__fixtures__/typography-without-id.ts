export const typographyWithoutId = {
  "style": [
      {
          "deviceId": "desktop",
          "value": {
              "fontFamily": "Nanum Gothic Coding",
              "fontSize": {
                  "unit": "px",
                  "value": 18
              },
              "fontWeight": 400,
              "lineHeight": 1.5
          }
      }
  ]
}
