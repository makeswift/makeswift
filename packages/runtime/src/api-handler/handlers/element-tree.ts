import { ReactRuntimeCore } from '../../react'
import { type Element } from '../../state/react-page'
import { type ApiRequest, ApiResponse, type ErrorResponseBody } from '../request-response'

export async function elementTreeHandler(
  req: ApiRequest,
  { runtime }: { runtime: ReactRuntimeCore },
): Promise<ApiResponse<{ elementTree: Element } | ErrorResponseBody>> {
  const body = await req.json()
  const { elementTree, replacementContext } = body

  if (elementTree == null) {
    return ApiResponse.json({ message: 'elementTree must be defined' }, { status: 400 })
  }

  if (replacementContext == null) {
    return ApiResponse.json({ message: 'replacementContext must be defined' }, { status: 400 })
  }

  const generatedElementTree = runtime.copyElementTree(elementTree, replacementContext)
  return ApiResponse.json({ elementTree: generatedElementTree })
}
