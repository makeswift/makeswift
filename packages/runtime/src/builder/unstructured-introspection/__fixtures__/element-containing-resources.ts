import { Element } from '@makeswift/controls'

/*
  Below is a fairly minimal example of an Element containing one of each:
    - Swatch
    - File
    - Typography
    - Global Element
    - Page Link (called 'Page' in APIResourceType)
    - Table
*/
export const elementContainingResources: Element = {
  key: '2c5e9ffc-dbcf-486a-ac62-8ac9efa539f0',
  props: {
    children: {
      '@@makeswift/type': 'prop-controllers::grid::v1',
      value: {
        columns: [
          {
            deviceId: 'desktop',
            value: {
              spans: [[12], [12], [12], [12], [12], [12]],
              count: 12,
            },
          },
        ],
        elements: [
          {
            key: 'cb66f6cb-0656-4424-81d7-0d03b27c45de',
            props: {
              children: {
                '@@makeswift/type': 'prop-controllers::grid::v1',
                value: {
                  columns: [
                    {
                      deviceId: 'desktop',
                      value: {
                        count: 12,
                        spans: [[12]],
                      },
                    },
                  ],
                  elements: [
                    {
                      key: '0f56d678-341a-4b05-bc9c-3a9f5bd7251e',
                      props: {
                        margin: [
                          {
                            deviceId: 'desktop',
                            value: {
                              marginBottom: {
                                unit: 'px',
                                value: 20,
                              },
                              marginLeft: 'auto',
                              marginRight: 'auto',
                              marginTop: null,
                            },
                          },
                        ],
                        text: {
                          descendants: [
                            {
                              children: [
                                {
                                  text: 'Resource: swatch only',
                                  typography: {
                                    style: [
                                      {
                                        deviceId: 'mobile',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 16,
                                          },
                                        },
                                      },
                                      {
                                        deviceId: 'desktop',
                                        value: {
                                          color: {
                                            alpha: 1,
                                            swatchId:
                                              'U3dhdGNoOmQ1ZWU3MmY1LThmMTQtNDQzZC05YmQ1LTJiMDA1MTIyYTA5NQ==',
                                          },
                                          fontSize: {
                                            unit: 'px',
                                            value: 18,
                                          },
                                          fontWeight: 400,
                                          lineHeight: 1.5,
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                              type: 'default',
                            },
                          ],
                          key: 'c632090b-2269-40c9-8924-0ecc5effa651',
                          type: 'makeswift::controls::rich-text-v2',
                          version: 2,
                        },
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              unit: 'px',
                              value: 700,
                            },
                          },
                        ],
                      },
                      type: './components/Text/index.js',
                    },
                  ],
                },
              },
              padding: [
                {
                  deviceId: 'desktop',
                  value: {
                    paddingBottom: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingLeft: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingRight: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingTop: {
                      unit: 'px',
                      value: 10,
                    },
                  },
                },
              ],
            },
            type: './components/Box/index.js',
          },
          {
            key: 'cf3e1cd1-e1bb-4420-8644-34d780b5465b',
            props: {
              children: {
                '@@makeswift/type': 'prop-controllers::grid::v1',
                value: {
                  columns: [
                    {
                      deviceId: 'desktop',
                      value: {
                        count: 12,
                        spans: [[6, 6]],
                      },
                    },
                  ],
                  elements: [
                    {
                      key: 'c42ddde2-b9ff-4787-909f-d23866ca9130',
                      props: {
                        file: {
                          '@@makeswift/type': 'prop-controllers::image::v2',
                          value: {
                            id: 'RmlsZTpjNzc2MjI3ZC1iNmZjLTQ3MGYtYTY1OC00NWQ4NzA2NDVkMDM=',
                            type: 'makeswift-file',
                            version: 1,
                          },
                        },
                        margin: {
                          '@@makeswift/type': 'prop-controllers::margin::v1',
                          value: [
                            {
                              deviceId: 'desktop',
                              value: {
                                marginLeft: {
                                  unit: 'px',
                                  value: 270,
                                },
                                marginRight: 'auto',
                              },
                            },
                          ],
                        },
                        width: {
                          '@@makeswift/type': 'prop-controllers::width::v1',
                          value: [
                            {
                              deviceId: 'desktop',
                              value: {
                                unit: '%',
                                value: 15,
                              },
                            },
                          ],
                        },
                      },
                      type: './components/Image/index.js',
                    },
                    {
                      key: 'ce4edb81-8832-4266-a152-4e506e1e754d',
                      props: {
                        margin: {
                          '@@makeswift/type': 'prop-controllers::margin::v1',
                          value: [
                            {
                              deviceId: 'desktop',
                              value: {
                                marginBottom: {
                                  unit: 'px',
                                  value: 20,
                                },
                                marginLeft: 'auto',
                                marginRight: 'auto',
                                marginTop: null,
                              },
                            },
                          ],
                        },
                        text: {
                          descendants: [
                            {
                              children: [
                                {
                                  text: 'Resource: File only',
                                  typography: {
                                    style: [
                                      {
                                        deviceId: 'desktop',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 18,
                                          },
                                          fontWeight: 400,
                                          lineHeight: 1.5,
                                        },
                                      },
                                      {
                                        deviceId: 'mobile',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 16,
                                          },
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                              type: 'default',
                            },
                          ],
                          key: '061f43b1-72aa-46b6-b429-c4a55eac09c5',
                          type: 'makeswift::controls::rich-text-v2',
                          version: 2,
                        },
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              unit: 'px',
                              value: 700,
                            },
                          },
                        ],
                      },
                      type: './components/Text/index.js',
                    },
                  ],
                },
              },
              padding: [
                {
                  deviceId: 'desktop',
                  value: {
                    paddingBottom: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingLeft: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingRight: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingTop: {
                      unit: 'px',
                      value: 10,
                    },
                  },
                },
              ],
            },
            type: './components/Box/index.js',
          },
          {
            key: 'ceebe1f0-31c1-4314-b387-df463d076751',
            props: {
              children: {
                '@@makeswift/type': 'prop-controllers::grid::v1',
                value: {
                  columns: [
                    {
                      deviceId: 'desktop',
                      value: {
                        count: 12,
                        spans: [[12]],
                      },
                    },
                  ],
                  elements: [
                    {
                      key: '87ab03ef-3a10-49d9-b596-719834f894dd',
                      props: {
                        margin: {
                          '@@makeswift/type': 'prop-controllers::margin::v1',
                          value: [
                            {
                              deviceId: 'desktop',
                              value: {
                                marginBottom: {
                                  unit: 'px',
                                  value: 20,
                                },
                                marginLeft: 'auto',
                                marginRight: 'auto',
                                marginTop: null,
                              },
                            },
                          ],
                        },
                        text: {
                          descendants: [
                            {
                              children: [
                                {
                                  text: 'Resource: typography only',
                                  typography: {
                                    id: 'VHlwb2dyYXBoeTo1ODVkMGE1ZS0zYmUyLTQyNzEtOTczYS1mNWNjN2UzMzk4ODg=',
                                    style: [],
                                  },
                                },
                              ],
                              type: 'default',
                            },
                          ],
                          key: '9c7f5fa6-7f15-4e60-91fd-778827ab3449',
                          type: 'makeswift::controls::rich-text-v2',
                          version: 2,
                        },
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              unit: 'px',
                              value: 700,
                            },
                          },
                        ],
                      },
                      type: './components/Text/index.js',
                    },
                  ],
                },
              },
              padding: [
                {
                  deviceId: 'desktop',
                  value: {
                    paddingBottom: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingLeft: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingRight: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingTop: {
                      unit: 'px',
                      value: 10,
                    },
                  },
                },
              ],
            },
            type: './components/Box/index.js',
          },
          {
            key: '6f2a2437-f690-4644-84d6-14073f1404d4',
            props: {
              children: {
                '@@makeswift/type': 'prop-controllers::grid::v1',
                value: {
                  columns: [
                    {
                      deviceId: 'desktop',
                      value: {
                        count: 12,
                        spans: [[6, 6]],
                      },
                    },
                  ],
                  elements: [
                    {
                      key: '42a6048c-ea36-41a4-9783-9c28f58fa3e6',
                      props: {
                        margin: [
                          {
                            deviceId: 'desktop',
                            value: {
                              marginBottom: {
                                unit: 'px',
                                value: 20,
                              },
                              marginLeft: 'auto',
                              marginRight: 'auto',
                              marginTop: null,
                            },
                          },
                        ],
                        text: {
                          descendants: [
                            {
                              children: [
                                {
                                  text: 'Resource: Global Element only',
                                  typography: {
                                    style: [
                                      {
                                        deviceId: 'desktop',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 18,
                                          },
                                          fontWeight: 400,
                                          lineHeight: 1.5,
                                        },
                                      },
                                      {
                                        deviceId: 'mobile',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 16,
                                          },
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                              type: 'default',
                            },
                          ],
                          key: '09c570fa-7d82-4886-8c33-ad418202d562',
                          type: 'makeswift::controls::rich-text-v2',
                          version: 2,
                        },
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              unit: 'px',
                              value: 700,
                            },
                          },
                        ],
                      },
                      type: './components/Text/index.js',
                    },
                    {
                      key: '229fc932-2fdf-46eb-8ff6-ad31d3660d0a',
                      type: 'reference',
                      value: 'R2xvYmFsRWxlbWVudDphOTRjYjM3ZS02OWM2LTQ1OGQtODIxNS1mZTA2NGQ3OWZmNjI=',
                    },
                  ],
                },
              },
              padding: [
                {
                  deviceId: 'desktop',
                  value: {
                    paddingBottom: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingLeft: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingRight: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingTop: {
                      unit: 'px',
                      value: 10,
                    },
                  },
                },
              ],
            },
            type: './components/Box/index.js',
          },
          {
            key: '3c83ae93-7cc9-47b1-8599-4efbd5bbd331',
            props: {
              children: {
                '@@makeswift/type': 'prop-controllers::grid::v1',
                value: {
                  columns: [
                    {
                      deviceId: 'desktop',
                      value: {
                        count: 12,
                        spans: [[6, 6]],
                      },
                    },
                  ],
                  elements: [
                    {
                      key: 'f1ee5bf3-ebec-4398-b4e1-73b13c8050dd',
                      props: {
                        margin: [
                          {
                            deviceId: 'desktop',
                            value: {
                              marginBottom: {
                                unit: 'px',
                                value: 20,
                              },
                              marginLeft: 'auto',
                              marginRight: 'auto',
                              marginTop: null,
                            },
                          },
                        ],
                        text: {
                          descendants: [
                            {
                              children: [
                                {
                                  text: 'Resource: Page Link only',
                                  typography: {
                                    style: [
                                      {
                                        deviceId: 'desktop',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 18,
                                          },
                                          fontWeight: 400,
                                          lineHeight: 1.5,
                                        },
                                      },
                                      {
                                        deviceId: 'mobile',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 16,
                                          },
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                              type: 'default',
                            },
                          ],
                          key: 'f3b4e0db-8141-4500-8b0d-19978cfcd664',
                          type: 'makeswift::controls::rich-text-v2',
                          version: 2,
                        },
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              unit: 'px',
                              value: 700,
                            },
                          },
                        ],
                      },
                      type: './components/Text/index.js',
                    },
                    {
                      key: '0ca675bf-a128-4a22-9cda-b4f07ad7b52e',
                      props: {
                        margin: [
                          {
                            deviceId: 'desktop',
                            value: {
                              marginBottom: {
                                unit: 'px',
                                value: 20,
                              },
                              marginLeft: 'auto',
                              marginRight: 'auto',
                              marginTop: null,
                            },
                          },
                        ],
                        text: {
                          descendants: [
                            {
                              children: [
                                {
                                  text: '',
                                },
                                {
                                  children: [
                                    {
                                      text: 'Sample Link to Page A',
                                      typography: {
                                        style: [
                                          {
                                            deviceId: 'desktop',
                                            value: {
                                              fontSize: {
                                                unit: 'px',
                                                value: 18,
                                              },
                                              fontWeight: 400,
                                              lineHeight: 1.5,
                                            },
                                          },
                                          {
                                            deviceId: 'mobile',
                                            value: {
                                              fontSize: {
                                                unit: 'px',
                                                value: 16,
                                              },
                                            },
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                  link: {
                                    payload: {
                                      openInNewTab: false,
                                      pageId:
                                        'UGFnZToxZWU1YmNjNi1mMGI1LTQxMGUtOWIzMy0wZTJjZTNkYTIwMzE=',
                                    },
                                    type: 'OPEN_PAGE',
                                  },
                                  type: 'link',
                                },
                                {
                                  text: '',
                                },
                              ],
                              type: 'default',
                            },
                          ],
                          key: '878ec264-dbb5-4b14-8dc3-651133ba9a8e',
                          type: 'makeswift::controls::rich-text-v2',
                          version: 2,
                        },
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              unit: 'px',
                              value: 700,
                            },
                          },
                        ],
                      },
                      type: './components/Text/index.js',
                    },
                  ],
                },
              },
              padding: [
                {
                  deviceId: 'desktop',
                  value: {
                    paddingBottom: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingLeft: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingRight: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingTop: {
                      unit: 'px',
                      value: 10,
                    },
                  },
                },
              ],
            },
            type: './components/Box/index.js',
          },
          {
            key: 'fac5fb69-c058-49a8-8b07-0917b2beddc8',
            props: {
              padding: [
                {
                  deviceId: 'desktop',
                  value: {
                    paddingBottom: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingLeft: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingRight: {
                      unit: 'px',
                      value: 10,
                    },
                    paddingTop: {
                      unit: 'px',
                      value: 10,
                    },
                  },
                },
              ],
              children: {
                '@@makeswift/type': 'prop-controllers::grid::v1',
                value: {
                  columns: [
                    {
                      deviceId: 'desktop',
                      value: {
                        spans: [[6, 6]],
                        count: 12,
                      },
                    },
                  ],
                  elements: [
                    {
                      key: '9274acff-aa56-4f04-b4f2-04e8106dc3a0',
                      props: {
                        margin: {
                          '@@makeswift/type': 'prop-controllers::margin::v1',
                          value: [
                            {
                              deviceId: 'desktop',
                              value: {
                                marginBottom: {
                                  unit: 'px',
                                  value: 20,
                                },
                                marginLeft: {
                                  unit: 'px',
                                  value: 0,
                                },
                                marginRight: 'auto',
                                marginTop: null,
                              },
                            },
                          ],
                        },
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              unit: 'px',
                              value: 700,
                            },
                          },
                        ],
                        text: {
                          type: 'makeswift::controls::rich-text-v2',
                          version: 2,
                          descendants: [
                            {
                              children: [
                                {
                                  text: 'Resource: table',
                                  typography: {
                                    style: [
                                      {
                                        deviceId: 'desktop',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 18,
                                          },
                                          fontWeight: 400,
                                          lineHeight: 1.5,
                                        },
                                      },
                                      {
                                        deviceId: 'mobile',
                                        value: {
                                          fontSize: {
                                            unit: 'px',
                                            value: 16,
                                          },
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                              type: 'default',
                            },
                          ],
                          key: '78307876-10bc-4240-a01d-071cd395a586',
                        },
                      },
                      type: './components/Text/index.js',
                    },
                    {
                      key: 'f5fbb12a-948b-414d-8267-b571713cbf60',
                      type: './components/Form/index.js',
                      props: {
                        gap: [
                          {
                            deviceId: 'desktop',
                            value: {
                              value: 10,
                              unit: 'px',
                            },
                          },
                        ],
                        width: [
                          {
                            deviceId: 'desktop',
                            value: {
                              value: 550,
                              unit: 'px',
                            },
                          },
                        ],
                        tableId: {
                          '@@makeswift/type': 'prop-controllers::table::v1',
                          value: 'VGFibGU6MDI2OTUwMDQtOThkMi00NDU1LThjNDAtYmFiMDU5YTY3NDQy',
                        },
                        fields: {
                          '@@makeswift/type': 'prop-controllers::table-form-fields::v1',
                          value: {
                            fields: [],
                            grid: [
                              {
                                deviceId: 'desktop',
                                value: {
                                  count: 12,
                                  spans: [[12]],
                                },
                              },
                            ],
                          },
                        },
                      },
                    },
                  ],
                },
              },
            },
            type: './components/Box/index.js',
          },
        ],
      },
    },
  },
  type: './components/Root/index.js',
}
