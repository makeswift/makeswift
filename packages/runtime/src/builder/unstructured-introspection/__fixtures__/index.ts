export * from './element-containing-resources'
