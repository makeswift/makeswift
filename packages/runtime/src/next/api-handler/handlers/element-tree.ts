import { NextApiRequest, NextApiResponse } from 'next'
import { ReactRuntime } from '../../../react'
import { Element } from '../../../state/react-page'
import { NextRequest, NextResponse } from 'next/server'
import { P, match } from 'ts-pattern'
import { type Context } from '../app-router-handler'

type ElementTreeResult = { elementTree: Element }

type ElementTreeError = { message: string }

export type ElementTreeResponse = ElementTreeResult | ElementTreeError

type ElementTreeHandlerArgs =
  | [request: NextRequest, context: Context, runtime: ReactRuntime]
  | [req: NextApiRequest, res: NextApiResponse<ElementTreeResponse>, runtime: ReactRuntime]

const routeHandlerPattern = [P.instanceOf(Request), P.any, P.any] as const
const apiRoutePattern = [P.any, P.any, P.any] as const

export default async function elementTree(
  request: NextRequest,
  context: Context,
  runtime: ReactRuntime,
): Promise<NextResponse<ElementTreeResponse>>
export default async function elementTree(
  req: NextApiRequest,
  res: NextApiResponse<ElementTreeResponse>,
  runtime: ReactRuntime,
): Promise<void>
export default async function elementTree(
  ...args: ElementTreeHandlerArgs
): Promise<NextResponse<ElementTreeResponse> | void> {
  const [, , runtime] = args

  const body = await match(args)
    .with(routeHandlerPattern, ([request]) => request.json())
    .with(apiRoutePattern, ([req]) => req.body)
    .exhaustive()
  const elementTree = body.elementTree
  const replacementContext = body.replacementContext

  if (elementTree == null) {
    const status = 400
    const body = { message: 'elementTree must be defined' }

    return match(args)
      .with(routeHandlerPattern, () => NextResponse.json(body, { status }))
      .with(apiRoutePattern, ([, res]) => res.status(status).json(body))
      .exhaustive()
  }

  if (replacementContext == null) {
    const status = 400
    const body = { message: 'replacementContext must be defined' }

    return match(args)
      .with(routeHandlerPattern, () => NextResponse.json(body, { status }))
      .with(apiRoutePattern, ([, res]) => res.status(status).json(body))
      .exhaustive()
  }

  const generatedElementTree = runtime.copyElementTree(elementTree, replacementContext)

  const response = { elementTree: generatedElementTree }

  return match(args)
    .with(routeHandlerPattern, () => NextResponse.json(response))
    .with(apiRoutePattern, ([, res]) => res.json(response))
    .exhaustive()
}
