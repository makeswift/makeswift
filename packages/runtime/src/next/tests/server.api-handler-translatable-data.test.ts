import { apiRequestFixtures } from './test-utils'

import { elementTree } from './__fixtures__/element-tree'

const PATH = '/api/makeswift/translatable-data'

afterEach(() => {
  jest.resetAllMocks()
})

describe('MakeswiftApiHandler', () => {
  describe.each(apiRequestFixtures)('[$router router] ${PATH}', ({ fixture }) => {
    test('does not require authentication', async () => {
      // Arrange
      const { testApiRequest } = fixture()

      // Act
      const { statusCode } = await testApiRequest({
        method: 'POST',
        path: PATH,
        body: {
          elementTree: {},
        },
      })

      // Assert
      expect(statusCode).toBe(200)
    })

    test.each([{ body: {}, message: 'elementTree must be defined.' }])(
      'returns 400 on missing arguments ($message)',
      async ({ body, message }) => {
        // Arrange
        const { testApiRequest } = fixture()

        // Act
        const { statusCode, jsonBody } = await testApiRequest({
          method: 'POST',
          path: PATH,
          body,
        })

        // Assert
        expect(statusCode).toBe(400)
        expect(await jsonBody).toEqual({ message })
      },
    )

    test('returns translatable data for the given element tree', async () => {
      // Arrange
      const { testApiRequest } = fixture()

      // Act
      const { statusCode, jsonBody } = await testApiRequest({
        method: 'POST',
        path: PATH,
        body: {
          elementTree,
        },
      })

      // Assert
      expect(statusCode).toBe(200)
      expect(await jsonBody).toMatchSnapshot('translatable data')
    })
  })
})
