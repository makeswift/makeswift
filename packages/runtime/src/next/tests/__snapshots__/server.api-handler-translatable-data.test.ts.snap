// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeswiftApiHandler [app router] \${PATH} returns translatable data for the given element tree: translatable data 1`] = `
{
  "translatableData": {
    "270019c4-4a80-483a-b701-757d4f7956a2:children": {
      "@@makeswift/type": "prop-controllers::text-input::v1",
      "value": "Shop now",
    },
    "2d5509fc-ef28-44c2-a52d-11278d893e45:text": {
      "0": "StyleMatch",
    },
    "4e25b048-88c4-4e2a-b74f-2161f2567ac3:text": {
      "0": "Why Shop With Us?",
    },
    "5350c1a7-0cfb-48ba-94b3-43014b3a01f9:text": {
      "0": "DropAlert Early Access",
    },
    "6b191363-f9c8-4988-9739-f74ca9cf7cee:text": {
      "0": "HUGE Summer Sale",
    },
    "7581e32e-b699-4d95-80c5-af532cbeef1d:text": {
      "0": "At our store, fashion meets innovation to create a shopping experience like no other. From our Virtual Try-On Studio that lets you preview outfits with precision, to StyleMatch that curates personalized looks just for you, we make finding your perfect fit effortless and fun. Our Fit Guarantee+ ensures every piece feels like it was made for you, while the Closet Companion App helps you build and style a wardrobe you’ll love. Plus, with our EcoEdit Filter, you can shop sustainably without compromising style. And with DropAlert Early Access, you’ll always stay one step ahead of the trends. It’s not just shopping—it’s styling made smart.",
    },
    "89ad1aad-73f7-4e99-962f-ebf7e05a6810:text": {
      "0": "Build your dream wardrobe digitally. Our Closet Companion App keeps track of what you own, helps you style new outfits, and suggests what to buy next to complete your look.",
    },
    "988e445b-cc28-46e0-b43d-eab4ed868f1a:text": {
      "0": "Be first in line for the hottest releases. With DropAlert™, you get exclusive early access to limited collections, back-in-stock alerts, and VIP-only styles—before anyone else.",
    },
    "9a55c845-b91b-4e65-a9bf-7647dff14836:text": {
      "0": "Closet Companion App",
    },
    "a80a02b2-0b2a-4719-8aff-7674f894e182:text": {
      "0": "Virtual Try-On Studio",
    },
    "aa3bb52a-350b-4bc3-9e80-aa6046f23472:altText": {
      "@@makeswift/type": "prop-controllers::text-input::v1",
      "value": "Night sky",
    },
    "bd59d14c-7d98-463f-8d8d-46ef53a765b5:text": {
      "0": "Fashion meets sustainability. Shop guilt-free by filtering our collection for ethically made, recycled, or low-impact pieces—because looking good should also feel good.",
    },
    "c712307f-4605-4204-9c30-f325c5861170:text": {
      "0": "Discover your perfect look in seconds. StyleMatch recommends pieces based on your taste, past purchases, and even your mood. It’s like having a personal stylist in your pocket.",
    },
    "cec3c603-3401-4b1b-8e2f-00d984f4ba71:text": {
      "0": "Never worry about the wrong size again. Fit Guarantee+ uses data-driven sizing plus free exchanges to make sure every piece fits you perfectly—or we’ll make it right, no questions asked.",
    },
    "dc706501-3b06-403c-929d-34048393d8f9:text": {
      "0": "EcoEdit Filter",
    },
    "e4c44cbd-8e06-4281-9a5a-a9c67998df1a:text": {
      "0": "See it. Style it. Love it. Our Virtual Try-On Studio lets you preview outfits on a 3D model that matches your body type—so you can shop with confidence and skip the dressing room drama.",
    },
    "ef06e719-b526-4fd7-95f7-324a0393df43:text": {
      "0": "Fit Guarantee+",
    },
  },
}
`;

exports[`MakeswiftApiHandler [pages router] \${PATH} returns translatable data for the given element tree: translatable data 1`] = `
{
  "translatableData": {
    "270019c4-4a80-483a-b701-757d4f7956a2:children": {
      "@@makeswift/type": "prop-controllers::text-input::v1",
      "value": "Shop now",
    },
    "2d5509fc-ef28-44c2-a52d-11278d893e45:text": {
      "0": "StyleMatch",
    },
    "4e25b048-88c4-4e2a-b74f-2161f2567ac3:text": {
      "0": "Why Shop With Us?",
    },
    "5350c1a7-0cfb-48ba-94b3-43014b3a01f9:text": {
      "0": "DropAlert Early Access",
    },
    "6b191363-f9c8-4988-9739-f74ca9cf7cee:text": {
      "0": "HUGE Summer Sale",
    },
    "7581e32e-b699-4d95-80c5-af532cbeef1d:text": {
      "0": "At our store, fashion meets innovation to create a shopping experience like no other. From our Virtual Try-On Studio that lets you preview outfits with precision, to StyleMatch that curates personalized looks just for you, we make finding your perfect fit effortless and fun. Our Fit Guarantee+ ensures every piece feels like it was made for you, while the Closet Companion App helps you build and style a wardrobe you’ll love. Plus, with our EcoEdit Filter, you can shop sustainably without compromising style. And with DropAlert Early Access, you’ll always stay one step ahead of the trends. It’s not just shopping—it’s styling made smart.",
    },
    "89ad1aad-73f7-4e99-962f-ebf7e05a6810:text": {
      "0": "Build your dream wardrobe digitally. Our Closet Companion App keeps track of what you own, helps you style new outfits, and suggests what to buy next to complete your look.",
    },
    "988e445b-cc28-46e0-b43d-eab4ed868f1a:text": {
      "0": "Be first in line for the hottest releases. With DropAlert™, you get exclusive early access to limited collections, back-in-stock alerts, and VIP-only styles—before anyone else.",
    },
    "9a55c845-b91b-4e65-a9bf-7647dff14836:text": {
      "0": "Closet Companion App",
    },
    "a80a02b2-0b2a-4719-8aff-7674f894e182:text": {
      "0": "Virtual Try-On Studio",
    },
    "aa3bb52a-350b-4bc3-9e80-aa6046f23472:altText": {
      "@@makeswift/type": "prop-controllers::text-input::v1",
      "value": "Night sky",
    },
    "bd59d14c-7d98-463f-8d8d-46ef53a765b5:text": {
      "0": "Fashion meets sustainability. Shop guilt-free by filtering our collection for ethically made, recycled, or low-impact pieces—because looking good should also feel good.",
    },
    "c712307f-4605-4204-9c30-f325c5861170:text": {
      "0": "Discover your perfect look in seconds. StyleMatch recommends pieces based on your taste, past purchases, and even your mood. It’s like having a personal stylist in your pocket.",
    },
    "cec3c603-3401-4b1b-8e2f-00d984f4ba71:text": {
      "0": "Never worry about the wrong size again. Fit Guarantee+ uses data-driven sizing plus free exchanges to make sure every piece fits you perfectly—or we’ll make it right, no questions asked.",
    },
    "dc706501-3b06-403c-929d-34048393d8f9:text": {
      "0": "EcoEdit Filter",
    },
    "e4c44cbd-8e06-4281-9a5a-a9c67998df1a:text": {
      "0": "See it. Style it. Love it. Our Virtual Try-On Studio lets you preview outfits on a 3D model that matches your body type—so you can shop with confidence and skip the dressing room drama.",
    },
    "ef06e719-b526-4fd7-95f7-324a0393df43:text": {
      "0": "Fit Guarantee+",
    },
  },
}
`;
